lib/types/noteData.js

```diff
@@ -0,0 +1,34 @@
+/*!
+ * Tozny e3db
+ *
+ * LICENSE
+ *
+ * Tozny dual licenses this product. For commercial use, please contact
+ * info@tozny.com. For non-commercial use, this license permits use of the
+ * software only by government agencies, schools, universities, non-profit
+ * organizations or individuals on projects that do not receive external
+ * funding other than government research grants and contracts. Any other use
+ * requires a commercial license. For the full license, please see LICENSE.md,
+ * in this source repository.
+ *
+ * @copyright Copyright (c) 2018-19 Tozny, LLC (https://tozny.com)
+ */
+
+'use strict'
+
+import { default as Signable } from '../types/signable'
+
+/**
+ * Representation of either plaintext or encrypted data encapsulated in a record.
+ */
+export default class NoteData extends Signable {
```

Comment:
could the 'data' sub-type actually be shared between records and notes?

Reply:
Yes, but I don't see the value of having NoteData and RecordData point to a Data object. I would like to remove RecordData/NoteData to just have Data, but wasn't sure if removing it was the right way to go

Reply:
that I would like to see as well. The advantage is it helps enforce consistency between records data and note data -- which is purposefully written to be identical.